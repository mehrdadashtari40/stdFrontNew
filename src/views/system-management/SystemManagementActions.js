import getAuthonticatedJSON from '../../common/utils/functions/getAuthenticatedJSON';

export const FETCH_CATEGORIES = 'FETCH_CATEGORIES';
export const FETCH_CATEGORY = 'FETCH_CATEGORY';
export const LOAD_USER = 'LOAD_USER';
export const LOAD_GROUPS = 'LOAD_GROUPS';
export const LOAD_DEPARTMENTS = 'LOAD_DEPARTMENTS';
export const LOAD_DEPARTMENT = 'LOAD_DEPARTMENT';



export function loadCategories(url) {
	return dispatch => {
		return getAuthonticatedJSON(url).then(response => {
			dispatch(setCategories(response));
		});
	};
}

export function setCategories(data) {
	return {
		type: FETCH_CATEGORIES,
		Categories: data,
	};
}

export function loadCategory(url) {
	return dispatch => {
		return getAuthonticatedJSON(url).then(response => {
			dispatch(setCategory(response));
		});
	};
}

export function setCategory(data) {
	return {
		type: FETCH_CATEGORY,
		Category: data,
	};
}

export function loadUser(url) {
	return dispatch => {
		return getAuthonticatedJSON(url).then(response => {
			dispatch(setUser(response));
		});
	};
}

export function setUser(data) {
	return {
		type: LOAD_USER,
		User: data,
	};
}

export function loadGroups(url) {
	return dispatch => {
		return getAuthonticatedJSON(url).then(response => {
			dispatch(setGroups(response));
		});
	};
}

export function setGroups(data) {
	return {
		type: LOAD_GROUPS,
		Groups: data,
	};
}

export function loadDepartments(url) {
	return dispatch => {
		return getAuthonticatedJSON(url).then(response => {
			dispatch(setDepartments(response));
		});
	};
}

export function setDepartments(data) {
	return {
		type: LOAD_DEPARTMENTS,
		Departments: data,
	};
}


export function loadDepartment(url) {
	return dispatch => {
		return getAuthonticatedJSON(url).then(response => {
			dispatch(setDepartment(response));
		});
	};
}

export function setDepartment(data) {
	return {
		type: LOAD_DEPARTMENT,
		Department: data,
	};
}


