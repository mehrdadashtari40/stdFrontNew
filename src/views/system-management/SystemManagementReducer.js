import {
	FETCH_CATEGORIES,
	FETCH_CATEGORY,
	LOAD_USER,
	LOAD_GROUPS,
	LOAD_DEPARTMENTS,
	LOAD_DEPARTMENT,
} from './SystemManagementActions';

const initialState = {
	Categories: [],
	Category: [],
	User: [],
	Users: [],
	Groups: [],
	Departments: [],
	Department: [],
};

export function systemManagementReducer(state = initialState, action) {
	switch (action.type) {
		case FETCH_CATEGORIES:
			return {
				...state,
				Categories: action.Categories,
			};

		case FETCH_CATEGORY:
			return {
				...state,
				Category: action.Category,
			};

		case LOAD_USER:
			return {
				...state,
				User: action.User,
			};
		case LOAD_GROUPS:
			return {
				...state,
				Groups: action.Groups,
			};
		case LOAD_DEPARTMENTS:
			return {
				...state,
				Departments: action.Departments,
			};
		case LOAD_DEPARTMENT:
			return {
				...state,
				Department: action.Department,
			};

		default:
			return state;
	}
}
