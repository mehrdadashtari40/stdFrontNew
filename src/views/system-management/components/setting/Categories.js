import React from 'react'
import Form from "../../utils/CategoryForm";
import {config}  from '../../../../config/config';
import $ from 'jquery';
import getAuthonticatedJSON from '../../../../common/utils/functions/getAuthenticatedJSON';
import { connect } from 'react-redux';
import * as Actions from '../../SystemManagementActions';
import { withTranslation } from 'react-i18next';
import {compose} from 'redux';
import MaterialDataTable from "../../../../common/tables/components/MaterialDataTable";
import CardContent from "@mui/material/CardContent";
import Card from "@mui/material/Card";
import {smallBox, SmartMessageBox} from "../../../../common/utils/functions";
import deleteAuthonticatedJSON from "../../../../common/utils/functions/deleteAuthonticatedJSON";
 class Categories extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			Edit: false,
			Show: [],
			refresh: false,
		};
		this.handelIndex = this.handelIndex.bind(this);
		this.handelCreate = this.handelCreate.bind(this);
		this.handelShow = this.handelShow.bind(this);
		this.handelDelete = this.handelDelete.bind(this);
	}
	componentDidMount() {
		let self = this;
		this.handelIndex();
		$('#CategoryBox').on('hidden.bs.modal', function () {
			self.handelIndex();
			// document.getElementById("new-role-form").reset();
		})

	}
	handelIndex() {
		this.props.loadCategories(config.apiServer + 'project/categories');
		this.setState({Edit: false});
	}
	handelCreate() {
		$("input[name=cat_uid]").val("")
		$("input[name=cat_name]").val("")
		$('#CategoryBox').modal('show');
	}
	handelShow(data) {
		getAuthonticatedJSON(config.apiServer + 'project/category/' + data['cat_uid'],
			{
				headers: {
					"Authorization": "Bearer " + localStorage.getItem('access_token'),
					'Content-Type': 'application/json'
				}
			}).then((res) => {
			this.setState({Show: res});
			$('#CategoryBox').modal('show');
			this.SetData();
			// this.disableFields();

		});

	}
	handelDelete(item){
		 let self = this;
		 SmartMessageBox({
			 title: this.props.t('Delete confirm title'),
			 content: this.props.t('Delete confirm content message'),
			 buttons: '[' + this.props.t('No') + '][' + this.props.t('Yes') + ']'
		 }, (ButtonPressed) => {
			 if (ButtonPressed === this.props.t('Yes')) {
				 let catId = item.cat_uid;
				 deleteAuthonticatedJSON(config.apiServer + 'project/category/'+catId).then((res)=>{
					 smallBox({
						 title: this.props.t('Successfully'),
						 content: "<i>"+this.props.t('The category have been deleted successfully')+"</i>",
						 color: "#659265",
						 iconSmall: "fa fa-check fa-2x fadeInRight animated",
						 timeout: 4000
					 });
					 $('#CategoryBox').modal('hide');
					 self.props.setCategories(self.props.Categories.filter(x=>x.cat_uid !== item.cat_uid));
				 }).catch(error=>{
				 	if(error.responseJSON.error !== undefined){
						smallBox({
							title: this.props.t('Failure'),
							content: "<i>"+this.props.t(error.responseJSON.error.message)+"</i>",
							color: "#af0808",
							iconSmall: "fa fa-check fa-2x fadeInRight animated",
							timeout: 4000
						});
					}

				 });
			 }
		 });
	 }
	SetData() {
		let data = this.state.Show;
		$("input[name=cat_uid]").val(data['cat_uid']);
		$("input[name=cat_name]").val(data['cat_name']);
	}
    render() {
		let self = this;
        return (
            <>
				<Card id="content">
					<CardContent>
						<h5 className={'table-header-title'}>
							<i className={"fal fa-cogs table-header-icon"}></i>
							{this.props.t('Process categories')}
							<button className="btn btn_inbox" onClick={this.handelCreate}>
								<i className="fa fa-plus" />
								<span>{this.props.t('Add New Category')}</span>
							</button>
						</h5>

						<div className="row">
							<div className="col-sm-12 col-md-12 col-lg-12 p-0">
								<MaterialDataTable
									items={this.props.Categories}
									columns={[
										{
											dataField: 'cat_name',
											text: this.props.t('Category Title'),
											filterable: 'cat_name',
											formatter: function (cell, item) {
												return <div className="text-primary" data-id={item.id}><b>{item.cat_name}</b></div>
											}
										},
										{
											dataField: 'cat_total_processes',
											text: this.props.t('PROCESS Count'),
											filterable: 'cat_total_processes'
										},
										{
											dataField: 'actions',
											text: this.props.t('actions'),
											formatter: function (cell, item) {
												return (<div className={'table-actions'}>
													<i className={'fas fa-edit'}
													   onClick={() => self.handelShow(item)}
													/>
													<i className={'fas fa-trash'}
													   onClick={() => self.handelDelete(item)}
													/>
												</div>)
											}
										},
									]}
								/>
							</div>
						</div>
					</CardContent>
				</Card>
                <div className="modal fade-in-up" id="CategoryBox" role="dialog" >
                    <div className="modal-dialog">
                        <div className="modal-content" id="CategoryBoxBody">
                            <Form  Refresh={this.handelIndex} Edit={this.state.Edit} Data={this.state.Show}/>
                        </div>
                    </div>
                </div>
            </>
        )
	}
}

export default compose( withTranslation(),connect(state => state.systemManagement, Actions))(Categories);
