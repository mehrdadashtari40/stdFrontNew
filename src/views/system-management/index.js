import Categories from './components/setting/Categories';
import PmTables from './components/setting/PmTables';
import Backup from './components/backup/Backup';
import Setting from './components/setting/Setting';
import Users from './components/users/Users';
import Groups from './components/users/Groups';
import Department from './components/users/Departments';
export const routes = [
	{
		path: '/process-categories',
		exact: true,
		component: Categories,
		name: 'Categories',
	},
	{
		path: '/pm-tables',
		exact: true,
		component: PmTables,
		name: 'PmTables',
	},
	{
		path: '/backup',
		exact: true,
		component: Backup,
		name: 'Backup',
	},
	{
		path: '/setting',
		exact: true,
		component: Setting,
		name: 'Setting',
	},
	{
		path: '/users',
		exact: true,
		component: Users,
		name: 'Users',
	},
	{
		path: '/groups',
		exact: true,
		component: Groups,
		name: 'Groups',
  },
  
  {
		path: '/departments',
		exact: true,
		component: Department,
		name: 'Department',
	},
];
export * from './SystemManagementActions';
export * from './SystemManagementReducer';
