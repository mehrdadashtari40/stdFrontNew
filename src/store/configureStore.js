import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import { composeWithDevTools } from 'redux-devtools-extension';

import { userReducer, requestUserInfo } from '../common/user';
import { i18nReducer, i18nInit } from '../common/i18n';
import { trelloReducer } from '../views/trello';
import { systemManagementReducer } from '../views/system-management';
import { ProcessManagementReducer } from '../views/process-management';

import {reducer as layout2Reducer} from '../common/layout/_redux/layoutRedux'
import {reducer as tableReducer} from '../../src/common/tables/_redux/tableRedux'


import {reducer as basicInformationReducer} from '../views/basic-information/_redux/baseInfoRedux';
import LogsReducer from '../views/logs/LogReducesrs';

import BIReducer from '../views/BI/BIReducer';
import {reducer as inboxReducer} from "../views/inbox/_redux/inboxRedux";
import {reducer as authReducer} from '../../src/views/auth/_redux/authRedux'

import {reducer as ServiceDeskReducer} from "../views/service-desk/_redux/serviceDeskRedux";
import {reducer as SupervisorReducer} from "../views/supervisor/_redux/supervisorRedux";
import {reducer as CustomSettingsReducer} from "../views/customSettings/_redux/customSettingsRedux";
import {reducer as MenuReducer} from "../views/menu/_redux/menuRedux";
import {reducer as WorkDeskReducer} from "../views/work-desk/_redux/workDeskRedux";

export const rootReducer = combineReducers({
	auth: authReducer,
	user: userReducer,
	i18n: i18nReducer,
	basicInformation: basicInformationReducer,
	processManagement: ProcessManagementReducer,
	systemManagement: systemManagementReducer,
	ServiceDesk:ServiceDeskReducer,
	BI:BIReducer,
	trello:trelloReducer,
	inbox:inboxReducer,
	layout2:layout2Reducer,
	table:tableReducer,
	logs:LogsReducer,
	supervisor:SupervisorReducer,
	customSettingsReducer:CustomSettingsReducer,
	menuReducer:MenuReducer,
	workDeskReducer:WorkDeskReducer
});

const store = createStore(
	rootReducer,
	applyMiddleware(thunk)
);

store.dispatch(requestUserInfo());

store.dispatch(i18nInit());

export default store;
