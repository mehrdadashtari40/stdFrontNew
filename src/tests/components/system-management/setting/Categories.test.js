import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import store from '../../../../store/configureStore';
import configureStore from 'redux-mock-store';
import Categories from '../../../../views/system-management/components/setting/Categories';
import * as selectActions from '../../../../views/system-management/SystemManagementActions';
import { I18nextProvider } from 'react-i18next';

// const store = mockStore();

jest.mock('react-i18next', () => ({
	withTranslation: () => Component => {
		Component.defaultProps = { ...Component.defaultProps, t: () => '' };
		return Component;
	},
}));


test('Make Snapshot file , Categories Component', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Categories />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	expect(wrapper).toMatchSnapshot();
});

it('invokes `componentDidMount` when mounted', () => {
	jest.spyOn(Categories.prototype, 'componentDidMount');
	shallow(
		<Provider store={store}>
			<Categories />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	expect(Categories.prototype.componentDidMount).toHaveBeenCalled();
	Categories.prototype.componentDidMount.mockRestore();
});



test('handelIndex Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
				<Categories />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.handelIndex).toBeDefined();
});

test('handelCreate Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
				<Categories />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.handelCreate).toBeDefined();
});

test('handelShow Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
				<Categories />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.handelShow).toBeDefined();
});

test('SetData Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Categories />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.SetData).toBeDefined();
});

test('disableFields Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Categories />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.disableFields).toBeDefined();
});

test('enableFields Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Categories />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.enableFields).toBeDefined();
});

describe('actions', () => {
	it('should create an action to loadCategories', () => {
		const data = 'test';
		const expectedAction = {
			type: selectActions.FETCH_CATEGORIES,
			Categories: data,
		};
		expect(selectActions.setCategories(data)).toEqual(expectedAction);
	});

	it('should create an action to loadCategory', () => {
		const data = 'test';
		const expectedAction = {
			type: selectActions.FETCH_CATEGORY,
			Category: data,
		};
		expect(selectActions.setCategory(data)).toEqual(expectedAction);
	});
});
