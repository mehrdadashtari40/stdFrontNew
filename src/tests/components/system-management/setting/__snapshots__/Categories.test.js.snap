// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make Snapshot file , Categories Component 1`] = `
<div
  id="content"
>
  <div
    className="row"
  >
    <BigBreadcrumbs
      className="col-xs-12 col-sm-7 col-md-7 col-lg-4"
      icon="fa fa-fw fa-category"
      items={
        Array [
          "دسته بندی",
          "مدیریت دسته بندی",
        ]
      }
    />
  </div>
  <WidgetGrid>
    <div
      className="row"
    >
      <article
        className="col-sm-12"
      >
        <JarvisWidget
          collapsed={false}
          color={false}
          colorbutton={false}
          custombutton={false}
          deletebutton={false}
          editbutton={false}
          fullscreenbutton={true}
          hidden={false}
          load={false}
          refresh={false}
          sortable={true}
          togglebutton={false}
        >
          <header>
            <span
              className="widget-icon"
            >
              <i
                className="fa fa-sitemap"
              />
               
            </span>
            <h2>
              فهرست دسته بندی ها
            </h2>
            <div
              className="widget-toolbar"
            >
              <a
                className="btn btn-outline-success little_btn"
                onClick={[Function]}
              >
                 
                <i
                  className="fa fa-plus"
                />
                <span>
                  مورد جدید
                </span>
              </a>
            </div>
          </header>
          <div
            className="tab-content"
            id="myTabContent1"
          >
            <div
              className="tab-pane fade in active"
            >
              <Datatable
                className="display projects-table table table-striped table-bordered table-hover"
                filter={true}
                fullTextSearch={false}
                options={
                  Object {
                    "columns": Array [
                      Object {
                        "class": "row-controller service-table-row service-id",
                        "data": null,
                        "render": [Function],
                      },
                      Object {
                        "class": "row-controller service-table-row",
                        "data": null,
                        "render": [Function],
                      },
                    ],
                    "data": Array [],
                    "iDisplayLength": 10,
                    "order": Array [
                      Array [
                        0,
                        "asc",
                      ],
                    ],
                  }
                }
                paginationLength={true}
                showAct={[Function]}
                width="100%"
              >
                <thead>
                  <tr>
                    <th
                      className="hasinput"
                      style={
                        Object {
                          "width": "22%",
                        }
                      }
                    >
                      <input
                        className="form-control "
                        id="dateselect_filter"
                        placeholder=""
                        type="text"
                      />
                    </th>
                    <th
                      className="hasinput"
                      style={
                        Object {
                          "width": "5%",
                        }
                      }
                    >
                      <input
                        className="form-control "
                        id="dateselect_filter"
                        placeholder=""
                        type="text"
                      />
                    </th>
                  </tr>
                  <tr>
                    <th
                      className="text-center"
                    />
                    <th
                      className="text-center"
                    />
                  </tr>
                </thead>
              </Datatable>
            </div>
          </div>
        </JarvisWidget>
      </article>
    </div>
  </WidgetGrid>
  <div
    className="modal fade-in-up"
    id="CategoryBox"
    role="dialog"
  >
    <div
      className="modal-dialog"
    >
      <div
        className="modal-content"
        id="CategoryBoxBody"
      >
        <CategoryForm
          Data={Array []}
          Disable={[Function]}
          Edit={false}
          Enable={[Function]}
          Refresh={[Function]}
          t={[Function]}
        />
      </div>
    </div>
  </div>
</div>
`;
